Handle K1 MPPA VLIW bundles in assembly

dkm opened the pull request:
If K1 MPPA compiler gets added someday (see https://github.com/mattgodbolt/compiler-explorer-image/pull/282 ), then this comment parsing tweak should be added.

K1 MPPA uses ;; as a bundle separator (same syntax as IA64,
if it gets added to CE someday...).
The bundle separator must be alone on the line, see for example:

```
        ld $r6 = 16[$r20]
        addx8d $r9 = $r2, $r5
        addw $r10 = $r2, -1
        ;;
        addd $r0 = $r6, $r9
        addd $r17 = $r10, 1
        cb.wlez $r2? .L3
        ;;
        copyd $r15 = $r0
        ;;
        maddd $r15 = $r17, -8
        ;;
```

Single ; or ;; followed by any non whitespace before eol still
considered a comment.

<!-- THIS COMMENT IS INVISIBLE IN THE FINAL PR, BUT FEEL FREE TO REMOVE IT
Thanks for taking the time to improve CE. We really appreciate it.
  Before opening the PR, please make sure that the tests & linter pass their checks,
  by running `make check`.
  In the best case scenario, you are also adding tests to back up your changes,
  but don't sweat it if you don't. We can discuss them at a later date.
Feel free to append your name to the CONTRIBUTORS.md file
Thanks again, we really appreciate this!
-->

partouf commented:
This breaks some unit tests, be sure to run 'npm run test' to check.

Bonuspoints if you can provide an example with these ;; for the unittests.

dkm commented:
On November 20, 2019 10:23:39 PM GMT+01:00, Patrick Quist <notifications@github.com> wrote:
>This breaks some unit tests, be sure to run 'npm run test' to check.
>
>Bonuspoints if you can provide an example with these ;; for the
>unittests.

ooops. I did "make test" and fixed one failing test. Will see why I did not see other failure ! I will add a new test also 😃

Thanks !

dkm commented:
The failing test has the following output:
```
      AssertionError: expected [ Array(16) ] to deeply equal [ Array(13) ]
      + expected - actual

       [
         {
           "source": [null]
      -    "text": ";"
      -  }
      -  {
      -    "source": [null]
      -    "text": ";"
      -  }
      -  {
      -    "source": [null]
      -    "text": ""
      -  }
      -  {
      -    "source": [null]
           "text": ".proc   _square: near"
         }
         {
           "source": [null]
```

I guess it's caused by the input having ';' alone on 2 lines:

```
1;
2; File generated by cc65 v 2.17 - Git f95481fa
3;
```

Is it really an error ? Looks to me that there are some text that should have been filtered ...

Is my understanding incorrect ?

mattgodbolt commented:
You're not wrong: looks like it ought to have been filtered before! Please feel free to update the expectation (there's some horrid "bless" code you can use in the test, or you can just hand-edit the json)

dkm commented:
Ok, thanks for proofreading this :D I'll update the testcase and add a now one for testing the new cases.

dkm commented:
In fact, the test was correct ! The lines beginning by `-` are what I've added and with `+` what has been removed... My regex was in fact not matching single ;. My reading was inverted, sorry about that. I've played a bit more on  regex101.com, I must admit that I still don't understand everything, but this should work. I still have to add a new test

The problem that way arise with this is for all assembly that really uses `;;` for comment, as in:
```
;; 
;; ======
;; this is bla
;; =======
;;
```
The `;;` alone on their lines will be displayed. This is correct for some exotic targets, but incorrect for most others.

Is this acceptable ? Or is there a need for filter that are target-dependent ?

dkm commented:
In fact, I had already an example of our assembly for something else (see 9784aa54bf6dd0e281294bea6ebb9e2b16bb7c14 ). So I've modified a bit this example and added the step for checking the comments filter.
I've cheated a bit, as in fact, a `;;` followed by a C-style comment `/* ... */` is also matched as a bundle separator and not a comment. We do not emit such assembly anymore, so I'm not sure it's worth the effort to support it.

mattgodbolt commented:
Sorry if took so long: merging now!
